hoffstadt/DearPyGui
# Get linux build fully working

---
name: Get Linux build fully working
about: Push building on linux to the same level as building on Windows and Mac, with fully separate debug and release configurations
title: ''
assignees: '@hoffstadt @Jah-On'

---




**Description:**
This makes building and testing on Linux very simple. For quick testing, one can build the sandbox like so (adjust `-j 6` to an appropriate amount of threads for your CPU):

```
./Scripts/BuildSandboxLinux.sh -j 6
```
To run it:
```
cd cmake-build-debug/DearSandbox; ./DearSandbox; cd ../..
```

To test and install the whole wheel:

```
./Scripts/BuildLocalWheelLinux.sh -j 6
```

To install it:

```
python3.9 -m pip install dist/dearpygui-0-cp39-cp39-manylinux1_x86_64.whl --force-reinstall
```

Afterwards, it can be used just like it was installed from pypi, making testing a breeze:
```
$ python3.9 
Python 3.9.0+ (default, Oct 20 2020, 08:43:38) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from dearpygui.core import *; from dearpygui.contexts import *; from dearpygui.simple import *;
>>> show_debug()
>>> start_dearpygui()
```

I tested this on the latest Manjaro and Ubuntu 20.04.

cpython is now built in `cpython/build/debug` or `cpython/build/release`. This is nice because it means the build won't show up as a change in the cpython submodule, but it should be tested properly to confirm it's working.

**Concerning Areas:**

1. To get the builds working properly, I moved the Python builds on Mac into `Dependencies/cpython/build` as well. Whether this worked has to be tested.
2. I tested this and it fully works when branched from c5772139, but after branching from 1caf2533 (split demo temporarily), I can't start the Sandbox:
```
cd cmake-build-debug/DearSandbox; ./DearSandbox; cd ../..
Traceback (most recent call last):
  File "../../DearSandbox/sandbox.py", line 3, in <module>
    import dearpygui.demo as demo
  File "../../DearPyGui/dearpygui/demo.py", line 7, in <module>
    import dearpygui.demo_widget as widgets
ModuleNotFoundError: No module named 'dearpygui.demo_widget'
```
3. Linux is a somewhat volatile ecosystem, so it would be great @Jah-On if you could pull this branch and test on your local machine as well.
